x/gethclone/main.go

```diff
@@ -0,0 +1,196 @@
+// The gethclone binary clones ethereum/go-ethereum Go packages, applying
+// semantic patches.
+package main
+
+import (
+	"context"
+	"fmt"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/token"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/subnet-evm/x/gethclone/astpatch"
+	"github.com/spf13/pflag"
+	"go.uber.org/multierr"
+	"golang.org/x/mod/modfile"
+	"golang.org/x/tools/go/ast/astutil"
+	"golang.org/x/tools/go/packages"
+
+	_ "embed"
+
+	// TODO(arr4n): change to using a git sub-module
+	_ "github.com/ethereum/go-ethereum/common"
+)
+
+func main() {
+	c := config{
+		astPatches: make(astpatch.PatchRegistry),
+	}
+
+	pflag.StringSliceVar(&c.packages, "packages", []string{"core/vm"}, `Geth packages to clone, with or without "github.com/ethereum/go-ethereum" prefix.`)
+	pflag.StringVar(&c.outputGoMod, "output_go_mod", "", "go.mod file of the destination to which geth will be cloned.")
+	pflag.Parse()
+
+	log.SetOutput(os.Stderr)
+	log.Print("START")
+	if err := c.run(context.Background()); err != nil {
+		log.Fatal(err)
+	}
+	log.Print("DONE")
+}
+
+type config struct {
+	// Externally configurable (e.g. flags)
+	packages    []string
+	outputGoMod string
+
+	// Internal
+	outputModule *modfile.Module
+	astPatches   astpatch.PatchRegistry
+
+	processed map[string]bool
+}
+
+const geth = "github.com/ethereum/go-ethereum"
```

Comment:
maybe consider moving this into a flag while retaining this value as the default value ?